### PR TITLE
P4TC Dash Issue fix

### PR DESCRIPTION
**Pull request by komaljai**
1. Support Type_varbit in EBPF.
2. Support Fields wider than 64 bits.
3. Support greater than 64 bit operations - Bitwise NOT operation
4. Handle istd.input_port in control_block keys.
5. Update pna.h file.
6. Remove packet resubmission from control block C file.
7. Support range, range-list and optional match-kinds.
8. Support annotation "ipv4" -
9. Add pass Eliminate Type_SerEnum to midend.

### CODE REVIEW

File: backends/tc/ebpfCodeGen.cpp [hunk 1]
```diff
@@ -1349,44 +1347,43 @@ void ControlBodyTranslatorPNA::processApply(const P4::ApplyMethod *method) {
             bool memcpy = false;
```
Comment (usha1830) [inline backends/tc/ebpfCodeGen.cpp:1331]: Can we change this variable name to something like "noEndiannessConversion". This conflicts with libfunc name and is confusing

File: testdata/p4tc_samples/test_ipv6_example.p4 [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+}
+
+header ipv6_t {
+    bit<4>   version;
+    bit<8>   trafficClass;
+    bit<20>  flowLabel;
+    bit<16>  payloadLength;
+    bit<8>   nextHeader;
+    bit<8>   hopLimit;
+    bit<128> srcAddr;
```
Comment (usha1830) [inline testdata/p4tc_samples/test_ipv6_example.p4]: Pls check if we need tctype annotation for ipv6 addresses as well

File: testdata/v1_1_samples/simple_router.p4 [hunk 3]
```diff
@@ -34,8 +34,8 @@ header_type ipv4_t {
         bit<8> ttl;
         bit<8> protocol;
         bit<16> hdrChecksum;
-        bit<32> srcAddr;
```
Comment (usha1830) [inline testdata/v1_1_samples/simple_router.p4]: Why is this test changed? This isn't tc case, so adding tc_type annotation doesn't look correct here

**Review by usha1830 — APPROVED** [review]
Overall looks fine. Please check some minor comments.

**Review by psivanup — APPROVED** [review]
Looks fine to me

**Pull request closed without merging**